serviceuser.go

```diff
@@ -401,12 +402,18 @@ func (su *ServiceUser) CFind(qrLevel QRLevel, filter []*dicom.Element) chan CFin
 			} else {
 				ch <- CFindResult{Elements: elems}
 			}
-			if resp.Status.Status != dimse.StatusPending {
-				if resp.Status.Status != 0 {
-					// TODO: report error if status!= 0
-					panic(resp)
-				}
-				break
+			switch resp.Status.Status {
+			case dimse.StatusPending:
+				continue receiveLoop
+			case dimse.StatusSuccess:
+				dicomlog.Vprintf(1, "dicom.serviceUser: C-MOVE received response success: %+v", elems)
+				ch <- CFindResult{Elements: elems}
```

Comment:
It seems that we sent the result twice to the channel in case `resp.Status.Status == dimse.StatusSuccess`, I saw the same code in line 403.

Also, could you help clarify in which case the server returns

* StatusPending
* StatusSuccess

It seems weird to me that the old code panics even in case of `dimse.StatusSuccess`.

Reply:
The old code panics the status is not Pending or Success.

https://dicom.nema.org/medical/dicom/current/output/chtml/part04/sect_cc.2.8.4.html

From the document, it seems that C-Find can returns multiple results, so status Pending indicates there are more results.